File: CS/WinExample.Module/BusinessObjects/DomainObject2.cs
using System;
using System.ComponentModel;

using DevExpress.Xpo;
using DevExpress.Data.Filtering;

using DevExpress.ExpressApp;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Persistent.Validation;

namespace WinExample.Module.BusinessObjects {
    [DefaultClassOptions]
    public class DomainObject2 : BaseObject {
        public DomainObject2(Session session)
            : base(session) { }
        private string _Name;
        public string Name {
            get { return _Name; }
            set { SetPropertyValue("Name", ref _Name, value); }
        }
        private DomainObject1 _DomainObject1;
        [Association("DomainObject1-DomainObject2s")]
        public DomainObject1 DomainObject1 {
            get { return _DomainObject1; }
            set { SetPropertyValue("DomainObject1", ref _DomainObject1, value); }
        }
    }

}

File: VB/WinExample.Module/BusinessObjects/DomainObject2.vb
Imports System
Imports System.ComponentModel

Imports DevExpress.Xpo
Imports DevExpress.Data.Filtering

Imports DevExpress.ExpressApp
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Persistent.Validation

Namespace WinExample.Module.BusinessObjects
    <DefaultClassOptions> _
    Public Class DomainObject2
        Inherits BaseObject

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub
        Private _Name As String
        Public Property Name() As String
            Get
                Return _Name
            End Get
            Set(ByVal value As String)
                SetPropertyValue("Name", _Name, value)
            End Set
        End Property
        Private _DomainObject1 As DomainObject1
        <Association("DomainObject1-DomainObject2s")> _
        Public Property DomainObject1() As DomainObject1
            Get
                Return _DomainObject1
            End Get
            Set(ByVal value As DomainObject1)
                SetPropertyValue("DomainObject1", _DomainObject1, value)
            End Set
        End Property
    End Class

End Namespace
